src/kixi/datastore/metadatastore/command_handler.clj

```diff
@@ -379,6 +379,43 @@ the generated 'update' specs.
                    ::ms/bundled-ids bundled-ids}
                   {:partition-key id}]))))
 
+(defmethod c/command-type->event-types
+  [:kixi.datastore/remove-files-from-bundle "1.0.0"]
+  [_]
+  #{[:kixi.datastore/files-removed-from-bundle "1.0.0"]
+    [:kixi.datastore/files-remove-from-bundle-rejected "1.0.0"]})
+
+(defn reject-remove-files-from-bundle
+  ([cmd reason id bundled-ids]
+   [{::event/type :kixi.datastore/files-remove-from-bundle-rejected
+     ::event/version "1.0.0"
+     :reason reason
+     ::ms/id id
+     ::ms/bundled-ids bundled-ids}
+    {:partition-key id}])
+  ([cmd reason id bundled-ids explain]
+   [{::event/type :kixi.datastore/files-remove-from-bundle-rejected
+     ::event/version "1.0.0"
+     :reason reason
+     ::ms/id id
+     ::ms/bundled-ids bundled-ids
+     :spec-explain explain}
+    {:partition-key id}]))
+
+(defn create-remove-files-from-bundle-handler
```

Comment:
I think something that can be abstract may appear at some point. I've been working on making the command handlers as similar in structure code wise as possible to try and get a sense of what that may be.

These last two are incredibly similar, others not so much.